contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -457,6 +461,76 @@ contract PrimaryAggregator is SiameseAggregatorBase, OCR2Abstract, OwnerIsCreato
     return latestAggregatorRoundId + 1;
   }
 
+  /**
+   *
+   * Section: Secondary Proxy
+   *
+   */
+
+  // Used to relieve stack pressure in transmit
+  struct Report {
+    int192 juelsPerFeeCoin;
+    uint32 observationsTimestamp;
+    bytes observers; // ith element is the index of the ith observer
+    int192[] observations; // ith element is the ith observation
+  }
+
+  // Transmission records the median answer from the transmit transaction at
+  // time timestamp
+  struct Transmission {
+    int192 answer; // 192 bits ought to be enough for anyone
+    uint32 observationsTimestamp; // when were observations made offchain
+    uint32 recordedTimestamp; // when was report received onchain
+  }
+  mapping(uint32 /* aggregator round ID */ => Transmission) internal s_transmissions;
+
+  // secondary proxy address, used to detect who's calling the contract methods
+  address internal immutable s_secondaryProxy;
+
+  // cutoff time defines the time window in which a secondary report is valid
+  uint32 internal s_cutoffTime;
+
+  /**
+   * @notice emitted when a new cutoff time is set
+   * @param cutoffTime the new defined cutoff time
+   */
+  event CutoffTimeSet(uint32 cutoffTime);
+
+  /**
+   * @notice sets the max time cutoff
+   * @param _cutoffTime new max cutoff timestamp
+   */
+  function setCutoffTime(uint32 _cutoffTime) external onlyOwner() {
+    s_cutoffTime = _cutoffTime;
+    emit CutoffTimeSet(s_cutoffTime);
+  }
+
+  /**
+   * @notice sync data with the primary rounds, return the freshest valid round id
+   */
+  function _getSyncPrimaryRound() internal view returns (uint80 roundId) {
+    // get the latest round id
+    uint32 latestAggregatorRoundId = s_hotVars.latestAggregatorRoundId;
+
+    // decreasing loop from the latest primary round id
+    for (uint80 round_ = latestAggregatorRoundId; round_ > 0; round_--) {
```

Comment:
Very small optimization, in for loops often see a pre-increment/decrement. In this loop, I would go with --round_